### ISSUE TITLE
Prepared Statements are not cashed? No speed improvement when name provided.

### ISSUE THREAD

**Issue by mjasnikovs** [brianc/node-postgres#1074]
For my experience node-postgres driver is wherry slow. So assuming, im doing something wrong.

I did test, with simple socket server, where i hacked pool and filed with prepared statements.

On start in node, spam pool with prepare statment, ignoring errors. Just to ensure all pool have value. max pools size * 2 
```JavaScript
Array(dbConfig.max*2).fill(() => {
	pool.connect((err, client, done) => {
		client.query('PREPARE getUserByEmail (text) AS select from users where email = $1;', err => done())
	})		
}).forEach(val => val())
``` 
I got raw ping speed, no delays at all when executing 1000 query's. ~35ms

```JavaScript
const time = new Date()
client.query(`EXECUTE getUserByEmail('${req.data.email}')`, (err, result) => {
	console.log((Math.round(new Date() - time)))
	if (err) throw new Error(err)
	res({}, 200)
	done()
})
```

I read about node-postgres, so to properly cash prepared statement, we have to provide name to query. 

```JavaScript
const time = new Date()
client.query({
	name: 'getUserByEmail',
	text: 'select * from users where email = $1',
	values: [req.data.email]
},(err, result) => {
	console.log((Math.round(new Date() - time)))
	if (err) throw new Error(err)
	res({}, 200)
	done()
})
``` 
But speed for such execution is 3x slower, then with hack. Around 80ms-150ms. This is huge performance lose.

**Comment by brianc** [comment]
I'd have to see the entire benchmark - general these kind of micro-benchmarks are missing some implementation detail or testing not exactly the right thing.  I don't have a lot of time to debunk them or point out where they might have gone wrong because I like to spend time fixing actual bugs or working on side-project stuff, but off the top of my head:

JIT warmup in v8
Query result parser "compiler" warmup in client
The first time a named query is dispatched it has to do the full `parse, bind, execute` phase, only the 2nd time does it use the name & skip the `parse` phase.
Calling `EXECUTE` in postgres results in a non-parameterized query which is less network packets and communication overhead in general

tl; dr - Lots of things could be causing this; its doubtful any of them are internal to the node-postgres codebase or easy to fix, but I'm open to a PR if you identify anything!

Thanks for submitting this anyway - good to know people are checking things out! 😄
